src/components/ui/customize/Radio/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+
+interface RadioProps {
+  label: string;
+  checked: boolean;
+  onClick?: (val: string | undefined) => void;
+  value: string | undefined;
+}
+
+const Radio = ({ label, checked, onClick, value }: RadioProps) => {
+  const [currentCheck, setCurrentCheck] = useState<boolean>(checked);
+  useEffect(() => {
```

Comment:
phần này k cần thiết, props checked sẽ tự update, k cần useffect

Reply:
Cái useEffect này phải cần để nó cập nhật trạng thái ông ơi, nếu ko có useEffect nó ko cập nhật được vì khi props thay đổi thì nó ko render lại

Reply:
vì sao props thay đổi? vì parent bị `re-render`. Khi state ở `parent` đổi, bản thân parent cũng sẽ dc `re-render`. Khi `parent` `re-render`, `children` cũng sẽ bị `re-render`. Và dùng `useEffect` sẽ làm chậm hơn vì React sẽ check state ở children 1 lần nữa khi điều đó k cần thiết. đây là ví dụ radio tui viết k dùng `useEffect` https://playcode.io/1605621. đọc thêm ở đây https://react.dev/learn/you-might-not-need-an-effect. Nếu tui có bỏ lỡ điều j trong trong code thì nói tui.